### DISCUSSION TITLE
How to work with Firebase Cloud Functions (2nd gen)?

### DISCUSSION THREAD

**Discussion by codyng** [fastify/fastify#5314] [General]
I have googled for 2 whole nights and still can't solve the problem.

I have tried to follow this: https://fastify.dev/docs/v4.15.x/Guides/Serverless/#google-cloud-functions
and this: https://firebase.google.com/docs/functions/http-events?gen=2nd (See Using existing Express or Flask apps)

None of the above docs can help me.

I am using latest Fastify and 2nd gen Firebase Cloud. (I didn't want to use Express although it was successful)

My code is as below:

```
const { onRequest } = require('firebase-functions/v2/https');
const Fastify = require('fastify');

let fastify;

const initFastify = async () => {

	fastify = Fastify({
		logger: false,
		ignoreTrailingSlash: true
	});

	fastify.addContentTypeParser('application/json', {}, (req, body, done) => {
		done(null, body.body);
	});

	fastify.get('/', async (request, reply) => {
		reply.send('Hello Fastify');
	});
};

initFastify();

exports.api = onRequest({
	region: 'asia-east2'
}, async (req, res) => {
	await fastify.ready();
	fastify.server.emit('request', req, res);
});
```

For local emulator, it's fine but it is failed to deploy to the Cloud:  "Error: There was an error deploying functions"

"Could not create or update Cloud Run service api, Container Healthcheck failed. Revision 'api-00002-yif' is not ready and cannot serve traffic. The user-provided container failed to start and listen on the port defined provided by the PORT=8080 environment variable. Logs for this revision might contain more information."

It's so paintful. Please help. ><

Thank you so so so much....

**Comment by metcoder95** [comment]
👋 
You create a `promise` out of `initFastify`, so `fastify` will be applied until the next tick but it seems that the function gets called almost immediately (within the same tick).

Did you tried moving the initialization also within the `onRequest`?

**Comment by codyng** [comment]
Thank you! I finally ended up solving with serverFactory + fastify.ready. Great! Thank you!
